lib/player.js

```diff
@@ -266,6 +266,21 @@ goog.require('shaka.util.StreamUtils');
  * @exportDoc
  */
 
+/**
+ * @event shaka.Player.SegmentParsedEvent
+ * @description Optionally fired from inside a manifest parser plugin if a newly
+ *   parsed segment has some extra metadata.
+ * @property {string} type
+ *   'segmentparsed'
+ * @property {string} mimeType
+ *   Mime type of the manifest parser.
+ * @property {number} streamId
+ *   Id of the stream that segment belonged to.
+ * @property {*} segmentData
+ *   Its format depends on the manifest parser.
```

Comment:
This seems particularly problematic to me.  For one, it's not obvious what the DASH equivalent is to the HLS type you're sending through here.  For another, the HLS type you're sending through here is an internal object, the format of which we should be free to change and the compiler should be free to rename.  So I wouldn't want to send that outside the Player to the app.

Reply:
I am still in the beginning of how DASH spec but, from my limited understanding, `emsg` is the equivalent in DASH. (I saw that `emsg` is handled separately in code right now). I would like to be corrected on this one. Moreover, given shaka does not intend to be restricted to HLS and DASH but intends to support any manifest, is it not good to provide a general purpose function for extracting custom data from the manifest?

Thank you for your time and looking forward to your thoughts.

Reply:
I completely agree with the fact that right now we are sending an internal object directly. I did not account for the fact that users of the method could possibly mutate the object.

Reply:
After thinking a little more about it, I realized that a user can always write their own parser if they have to read the custom data from the manifest. That exactly is something I wanted to avoid and thus came up with this PR.

Also, though `emsg` indicates segment specific details, it seems to be a feature of mp4 rather than DASH. Am I correct here?

Reply:
The 'emsg' box is an MP4 box, but it came out of the DASH spec if I remember correctly.  There is a schema specified in DASH that means "reload the manifest when you see this".  Shaka Player handles that directly, and passes any other schema up to the application.